velox/functions/prestosql/window/tests/NthValueTest.cpp

```diff
@@ -67,15 +85,39 @@ class NthValueTest : public WindowTestBase {
     // output row order in the tests.
     auto newOverClause = overClause_ + ", c4";
 
-    // The below tests cover nth_value invocations with constant and column
-    // arguments. The offsets could also give rows beyond the partition
-    // returning null in those cases.
-    WindowTestBase::testWindowFunction(
-        {vectors}, "nth_value(c4, 1)", {newOverClause}, kFrameClauses);
-    WindowTestBase::testWindowFunction(
-        {vectors}, "nth_value(c4, 7)", {newOverClause}, kFrameClauses);
-    WindowTestBase::testWindowFunction(
-        {vectors}, "nth_value(c4, c2)", {newOverClause}, kFrameClauses);
+    switch (valueFunctionType) {
+      case ValueFunctionType::kNthValue: {
```

Comment:
Switching on this enum is not needed. Please add a single testValueFunctions() functions that calls all the nth_value, first_value and last_value functions. Only that function can be called in all the unit tests.